### PR TITLE
Send email where a change is suggested

### PR DESCRIPTION
**Pull request by jakedaleweb**

### CODE REVIEW

**Comment by jakedaleweb** [comment]
TODO
Provide s3 graph in email

File: s3.go [hunk 1]
```diff
@@ -49,25 +54,38 @@ func createBucket(name string) {
...
+
+func getPresignedLink(location string, bucketName string) string {
+	sess := session.Must(session.NewSession(&aws.Config{Region: aws.String(region)}))
+	svc := s3.New(sess)
+
+	req, _ := svc.GetObjectRequest(&s3.GetObjectInput{
+		Bucket: aws.String(bucketName),
+		Key:    aws.String(location),
+	})
+	urlStr, err := req.Presign(1 * time.Hour)
```
Comment (stojg) [inline s3.go]: could we increase this a bit? 1hr feels short.. I only check my emails like daily

File: main.go [hunk 2]
```diff
@@ -142,15 +150,46 @@ func doStack(stackName, asgName, rdsName string) {
...
 	rPipe, wPipe := io.Pipe()
 	go func() {
 		writePlot(wPipe, p, 1024, 1024*(1/1.414))
 	}()
-	pipeToS3(rPipe, stackName)
+	imageLocation := pipeToS3(rPipe, stackName, bucketName)
+
+	if message != "" {
+		fmt.Println(message)
+		emailTeam("<p>" + message + "</p>" +
```
Comment (stojg) [inline main.go]: an upcoming improvement would probably to template this

Comment (stojg) [reply main.go]: do we want one email with aggregation or multiple? if it's more like daily reporting I would probably prefer a single email

File: main.go [hunk 3]
```diff
@@ -38,12 +41,15 @@ func main() {
...
+	bucketName := "mrhandy-graphs-" + os.Getenv("AWS_VAULT") + "-" + region
+	createBucket(bucketName)
+
 	var wg sync.WaitGroup
 	wg.Add(len(webGroups))
 	for _, webGroup := range webGroups {
 		go func(webGroup asg) {
 			defer wg.Done()
-			doStack(webGroup.Name, webGroup.AsgName, webGroup.RdsName)
+			doStack(webGroup.Name, webGroup.AsgName, webGroup.RdsName, bucketName)
```
Comment (stojg) [inline main.go]: we could possibly just pass in `webGroup` and `bucketname` to shorten the signature of `doStack` function..

**Pull request merged**